packages/dataviews/src/view-grid.js

```diff
@@ -32,94 +235,318 @@ function GridItem( {
+			)
+			.map( ( { itemId: _itemId } ) => _itemId );
+		return ids;
+	}, [ getGridState, id ] );
+
+	const getColumnItemIds = useCallback( () => {
+		const { renderedItems } = getGridState();
+		const [ index, rows ] = renderedItems.reduce(
+			( [ _index, _rows ], _item ) => {
+				const row = _rows[ _item.rowId ] || [];
+				row.push( _item.disabled ? undefined : _item.itemId );
+				_rows[ _item.rowId ] = row;
+				return [ _item.id === id ? row.length - 1 : _index, _rows ];
+			},
+			[ -1, {} ]
+		);
+		return Object.values( rows ).map( ( row ) => row[ index ] );
+	}, [ getGridState, id ] );
+
+	const keyDownHandler = useCallback(
+		( event ) => {
+			if ( event.defaultPrevented ) return;
+
+			const { target, currentTarget } = event;
+			const { key, ctrlKey, metaKey, shiftKey } = event;
+
+			const isChildEvent = target !== currentTarget;
+			const rtl = isRTL();
+			const movementMap = new Map( [
+				[ 'ArrowUp', up ],
+				[ 'ArrowDown', down ],
+				[ 'ArrowLeft', rtl ? next : previous ],
+				[ 'ArrowRight', rtl ? previous : next ],
+			] );
+
+			if ( movementMap.has( key ) ) {
+				const nextId = movementMap.get( key )();
+				if ( nextId ) {
+					setIsInteractive( false );
+					if ( isChildEvent ) {
+						move( nextId );
+					} else if ( shiftKey ) {
+						// Select this and 'next' grid item
+						const { itemId: nextItemId } = getGridItem( nextId );
+						setSelectedState( [ itemId, nextItemId ], true );
+					}
+				}
+				return;
+			}
+
+			if ( isChildEvent ) {
+				if ( key === 'Escape' || key === 'F2' ) {
+					// Focus grid item
+					event.preventDefault();
+					itemRef.current?.focus();
+				}
+
+				return;
+			}
+
+			if ( key === 'Enter' || key === 'F2' ) {
+				// Focus/action child item
+				event.preventDefault();
+				const action = key === 'Enter' ? 'click' : 'focus';
+				if ( isAppleOS() ? metaKey : ctrlKey ) {
+					// Cmd/ctrl operates on actions menu
+					itemActionsRef.current?.[ action ]();
+				} else {
+					// Everything else operates on primary field
+					primaryFieldRef.current?.[ action ]();
+				}
+				return;
+			}
+
+			if (
+				( key === 'a' || key === 'A' ) &&
+				( isAppleOS() ? metaKey : ctrlKey )
+			) {
+				// Select all/none
+				event.preventDefault();
+				setSelectedState(
+					data.map( ( _item ) => getItemId?.( _item ) ),
+					! shiftKey
+				);
+				return;
+			}
+
+			if ( key === ' ' ) {
+				event.preventDefault();
+				if ( ctrlKey ) {
+					// Select the entire column
+					setSelectedState( getColumnItemIds(), true );
+				} else if ( shiftKey ) {
+					// Select the entire row
+					setSelectedState( getRowItemIds(), true );
+				} else {
+					// Toggle selected state
+					toggleIsSelected();
+				}
+			}
+		},
+		[
+			data,
+			down,
+			getColumnItemIds,
+			getGridItem,
+			getItemId,
+			getRowItemIds,
+			itemId,
+			move,
+			next,
+			previous,
+			setSelectedState,
+			toggleIsSelected,
+			up,
+		]
+	);
+
+	const mediaEventHandlers = {
```

Comment:
I'm not sure we should be that opinionated here. It happens right now in our use case to have the media and primary field do the same, but that's not necessarily something everyone wants.